### DISCUSSION TITLE
Passing MSIINSTALLPERUSER=1 to MsiPackage installs the msi as Absent

### DISCUSSION THREAD

**Discussion by IvayloGugalov** [wixtoolset/issues#6394] [Questions]
Hello,

I have created a custom WiX Burn bootstrapper UI, which installs a MsiPackage from the Bundle element. The MsiPackage needs to reboot the system when installed, but I want it to be installed silently so it's ready for use immediately .

Passing `<MsiProperty Name="MSIINSTALLPERUSER" Value="1"/>` works fine and achieves the silent installation for the MsiPackage, but now, for some reason, my BootstrapperApplication asks for admin permissions before installation.

When trying to uninstall, I check the DetectBeginEventArgs - `Installed` flag and run `bootstrapper.Engine.Plan(LaunchAction.Uninstall);`, but this uninstalls only my BootstrapperApplication and leaves the MsiPackage installed.

Checking the DetectPackageCompleteEventArgs: `args.Absent == PackageState.Absent` is true. So when installed with `MSIINSTALLPERUSER=1` , the MsiPackage will be installed with state Absent instead of Present.

Is there a way to achieve the no restart installation and keep the PackageState of the msi file to Present?

**Comment by rseanhall** [comment]
Does the MsiPackage meet all of the requirements here: https://docs.microsoft.com/en-us/windows/win32/msi/msiinstallperuser? I don't understand the relationship between `MSIINSTALLPERUSER=1` and no restart installation.

**Reply by igugalov** [reply]
Most probably it doesn't. Installed in per-user context, the msi is ready to use, while in per-machine context, it requires a reboot to be fully operational. As I am not creating the MSI file, just using it my Wix project, I can just guess.

**Comment by robmen** [comment]
Burn does not support per-machine/per-user switching in MSIs today. Clear `ALLUSERS` property (or set `InstallScope='perUser'`) in the MSI package to make it per-user and the Bundle will go per-user to install that package.

**Reply by igugalov** [reply]
Thanks for the reply Rob. I will have to check other options or prompt for a system reboot after all, as I am not authoring the MSI, only wrapping it.

**Reply by robmen** [reply]
Does the MSI have `ALLUSERS` property set to `2`?

I'm not sure this will work but you could try adding a `MsiProperty` under your `MsiPackage` in your `Bundle` to set `ALLUSERS` to `""`. I do not remember if the tooling will pick that up and realize you're trying to force the package to per-user.

**Reply by igugalov** [reply]
I have tried setting `ALLUSERS=2`, but this does not affect the MSI. Setting it to `""` like this `<Variable Name="MyVariable" Value="" Type="string"/>`. This might work, but the MSI installation fails, as it can't initialize a dll inside it. The problem lays in the MSI file and probably I would need to take a different approach.

**Reply by rseanhall** [reply]
Can you tell us which MSI you're trying to use, or even upload it here so we can look at it? There seems to be a feature request here for you to be able to specify that you want an MSI that can either be per-user or per-machine to be forced as per-user when building the bundle.

**Reply by IvayloGugalov** [reply]
I have spoken with the manufacturer providing me the MSI and I am waiting on their response currently. If they are not able to resolve this I will come back to you.

**Reply by IvayloGugalov** [reply]
Hello again. I am trying to wrap my head around how does Burn decide in which context should the Bundle be build and installed. Is it decided at runtime or at build time, based on some property that is retrieved from the MSI directly or from the `MsiPackage` element? And if it's decided at build time, then setting `ForcePerMachine="no"` is only affecting the `MsiPackage`, but not the the Bundle itself? The goal is to set the Bundle in the same install context as the `MsiPackage` element inside it.

Also the `ALLUSERS` property is set to `2` inside the MSI file I am putting into my Bundle and the MSI is not created with Wix.

How do feature requests work with the Wix project as I wasn't able to find a way of working things out?

**Reply by rseanhall** [reply]
Have a look at https://github.com/wixtoolset/issues/issues/4914. No one has stepped up to implement it, probably because it's extremely complicated.
